src/mqt/predictor/ml/Predictor.py

```diff
@@ -141,22 +141,22 @@ def generate_compiled_circuits(
         if target_path is None:
             target_path = str(ml.helper.get_path_training_circuits_compiled())
 
-        source_circuits_list = []
-
-        for file in Path(source_path).iterdir():
-            if "qasm" in str(file):
-                source_circuits_list.append(str(file))
-
         path_zip = Path(source_path) / "mqtbench_training_samples.zip"
-        if len(source_circuits_list) == 0 and path_zip.exists():
+        if len(list(Path(source_path).glob("*.qasm"))) == 0 and path_zip.exists():
```

Comment:
If `source_path` contains lots of files, this might not perform well for a check that just tries to find out if a `.qasm` file is in there. The following might be a bit more performant.
```suggestion
        if not any(file.suffix == ".qasm" for file in Path(source_path).iterdir()) and path_zip.exists():
```

Reply:
Done.